frontend-web/webclient/app/Applications/Jobs/View.tsx

```diff
@@ -827,11 +862,104 @@ function parseUpdatesForAccess(updates: JobUpdate[]): ParsedSshAccess | null {
+                    (parsedTarget.type === "VNC" && isSupported(backendType, support, "vnc"));
+
+                if (parsedTarget.type === "WEB" && supportsInterface && job.status.state === "RUNNING") {
+                    const requests: OpenInteractiveSessionRequest[] = Array(job.specification.replicas)
+                        .fill(0).map((_, i) => ({sessionType: "WEB", id: job.id, rank: i, target: parsedTarget.target, port: parsedTarget.port}));
+
+                        try {
+                            const sessionResult = await invokeCommand<BulkResponse<InteractiveSession>>(JobsApi.openInteractiveSession(bulkRequestOf(...requests)));
+                            for (const res of sessionResult?.responses ?? []) {
+                                const webSession = (res.session as WebSession);
+                                result.unshift({
+                                    target: parsedTarget.target,
+                                    rank: webSession.rank,
+                                    type: "WEB",
+                                    link: webSession.redirectClientTo,
+                                });
+                            }
+                        } catch (e) {
+                            console.warn(e);
+                        }
+                } else if (parsedTarget.type === "VNC" && supportsInterface && job.status.state === "RUNNING") {
+                    result.unshift({
+                        target: parsedTarget.target,
+                        rank: parsedTarget.rank,
+                        type: "WEB",
+                        link: `/applications/vnc/${job.id}/${parsedTarget.rank}?hide-frame`
+                    });
+                }
+
+            }
+        }
+    }
+
+    const supportsInterface =
+        (appType === "WEB" && isSupported(backendType, support, "web")) ||
+        (appType === "VNC" && isSupported(backendType, support, "vnc"));
+
+    if (appType === "VNC" && supportsInterface) {
+        result.unshift({rank: 0, type: "VNC", link: `/applications/vnc/${job.id}/0?hide-frame`} as InterfaceTarget);
+    }
+
+    if (appType === "WEB" && supportsInterface && job.status.state === "RUNNING") {
+        const requests: OpenInteractiveSessionRequest[] = Array(job.specification.replicas)
+            .fill(0).map((_, i) => ({sessionType: "WEB", id: job.id, rank: i}));
+
+        try {
+            const resp = await invokeCommand<BulkResponse<InteractiveSession>>(JobsApi.openInteractiveSession(bulkRequestOf(...requests)));
+            for (const res of resp?.responses ?? []) {
+                const webSession = (res.session as WebSession);
+                result.unshift({
+                    rank: webSession.rank,
+                    type: "WEB",
+                    link: webSession.redirectClientTo
+                });
+            }
+        } catch (e) {
+            console.warn(e);
+        }
+    }
+
+    return result;
+}
+
```

Comment:
This function can be improved by not awaiting immediately after calling the backend, so both promises can be awaited on at the same time. Not an urgent fix, but a `TODO` could be added.